redaxo/src/core/lib/autoload.php

```diff
@@ -255,7 +255,7 @@ private static function _addDirectory($dir)
         $iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($dirPath, RecursiveDirectoryIterator::SKIP_DOTS));
         foreach ($iterator as $path => $file) {
             /** @var SplFileInfo $file */
-            if (!$file->isFile() || !in_array($file->getExtension(), ['php', 'inc'])) {
+            if (!in_array($file->getExtension(), ['php', 'inc']) || !$file->isFile()) {
```

Comment:
Ist nicht `is_file` gerade schnell? (https://github.com/redaxo/redaxo/pull/3556)
Hätte jetzt vermutet, dass `SplFileInfo::isFile` entsprechend auch schnell ist.

Habe es gerade mal angetestet:
```
$iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator(rex_path::base(), RecursiveDirectoryIterator::SKIP_DOTS));
$isFile = 0;
$checkExtension = 0;
foreach ($iterator as $file) {
    $t = microtime(true);
    $file->isFile();
    $isFile += microtime(true) - $t;

    $t = microtime(true);
    in_array($file->getExtension(), ['php', 'inc']);
    $checkExtension += microtime(true) - $t;
}
dump($isFile, $checkExtension);
```

Bekomme da ca. (bisschen schwanken die Zahlen natürlich, aber isFile ist bei mir immer schneller):
```
0.012999534606934 // isFile
0.013892412185669 // getExtension
```

Tausche ich in der foreach die Checks, also erst getExtension, dann isFile, sind die Unterschiede größer (es spielt also sicherlich noch ein File-Cache mit rein).
```
0.018749952316284 // getExtension
0.014780521392822 // isFile
```

Reply:
> Ist nicht is_file gerade schnell? (https://github.com/redaxo/redaxo/pull/3556

`is_file` ist schneller als `file_exists` aber da es eine IO operation ist, sollte es viel langsamer sein als jede string function

-----

meine these zu dem PR ist/war:

- `is_file` erzeugt IO auf der platte
- `$file->getExtension()` returned einen bekannten string auf den dann `in_array` läuft

-> IO ist in der regel viel langsamer

Reply:
anders gesagt: meine Änderung macht hauptsächlich dann sinn, wenn in dem zu iterienden ordner auch non-`.php` bzw. non-`.inc` dateien liegen